src/lib/device-types/index.ts

```diff
@@ -246,21 +313,32 @@ export const validateSlug = (slug?: string) => {
 	return slug;
 };
 
-export const deviceTypes = (): Promise<DeviceType[]> => {
-	return getDeviceTypes().then(deviceTypesInfos => {
-		// exclude aliases
-		return _(deviceTypesInfos)
-			.filter(
-				(deviceTypesInfo, slug) =>
-					deviceTypesInfo.latest.deviceType.slug === slug,
-			)
-			.map(deviceTypesInfo => deviceTypesInfo.latest.deviceType)
-			.value();
-	});
+export const deviceTypes = (api: PinejsClient): Promise<DeviceType[]> => {
+	return getDeviceTypes()
+		.then(deviceTypesInfos => {
+			// exclude aliases
+			return _(deviceTypesInfos)
+				.filter(
+					(deviceTypesInfo, slug) =>
+						deviceTypesInfo.latest.deviceType.slug === slug,
+				)
+				.map(deviceTypesInfo => deviceTypesInfo.latest.deviceType)
+				.value();
+		})
+		.then(deviceTypes => {
+			return getAccessibleSlugs(api).then((accessibleDeviceTypes: string[]) => {
+				return _.filter(deviceTypes, o => {
+					return _.includes(accessibleDeviceTypes, o.slug);
+				});
+			});
+		});
 };
 
-export const findBySlug = (slug: string): Promise<DeviceType> =>
-	deviceTypes()
+export const findBySlug = (
+	slug: string,
+	api: PinejsClient,
```

Comment:
Same here -- `api` should be the first argument.

And below in `normalizeDeviceType`, `getImageSize`, etc. Can you please ensure all changed functions accept `api` as their first argument?